rskj-core/src/test/java/co/rsk/remasc/RemascTestBlockStore.java

```diff
@@ -89,8 +93,35 @@ public void saveBlock(Block block, BigInteger cummDifficulty, boolean mainChain)
         ByteArrayWrapper wHash = wrap(block.getHash());
         blocks.add(block);
         hashIndex.put(wHash, block);
-        numberIndex.put(block.getNumber(), block);
+
+        if (mainChain)
+            numberIndex.put(block.getNumber(), block);
+
         totalDifficulty = totalDifficulty.add(block.getCumulativeDifficulty());
+
+        List<IndexedBlockStore.BlockInfo> blockInfos = index.get(block.getNumber());
+        if (blockInfos == null) {
+            blockInfos = new ArrayList<>();
+        }
+
+        IndexedBlockStore.BlockInfo blockInfo = null;
+        for (IndexedBlockStore.BlockInfo bi : blockInfos) {
+            if (areEqual(bi.getHash(), block.getHash())) {
```

Comment:
What's the difference between `org.spongycastle.util.Arrays#areEqual` and `java.util.Arrays#equals`?

Reply:
And `ByteUtil.fastEquals` too. We definitely need guidance here.

Reply:
To be totally honest i don't know the difference, i just used the one that i saw was the most used. Trying to respect convention.